src/stores/provider.ts

```diff
@@ -0,0 +1,154 @@
+import { capital } from 'case'
+import { defineStore } from 'pinia'
+import { computed, reactive, Ref, ssrRef } from '@nuxtjs/composition-api'
+import axios from 'axios'
+
+import {
+  AUDIO,
+  IMAGE,
+  SupportedMediaType,
+  supportedMediaTypes,
+} from '~/constants/media'
+import { warn } from '~/utils/console'
+import { useSearchStore } from '~/stores/search'
+
+import { providerServices } from '~/data/media-provider-service'
+import type { MediaProvider } from '~/models/media-provider'
+import { FetchState, useFetchState } from '~/composables/use-fetch-state'
+
+export interface ProviderState {
+  providers: {
+    audio: MediaProvider[]
+    image: MediaProvider[]
+  }
+  fetchState: {
+    audio: FetchState
+    image: FetchState
+  }
+}
+/**
+ * Sorts providers by their source_name property.
+ * @param data - initial unordered list of providers
+ */
+const sortProviders = (data: MediaProvider[]): MediaProvider[] => {
+  return [...data].sort((sourceObjectA, sourceObjectB) => {
+    const nameA = sourceObjectA.source_name.toUpperCase()
+    const nameB = sourceObjectB.source_name.toUpperCase()
+    return nameA.localeCompare(nameB)
+  })
+}
+/**
+ * Timestamp is used to limit the update frequency to one every 60 minutes per request.
+ */
+const lastUpdated: Ref<Date | null> = ssrRef(null)
+
+const updateFrequency = process.env.providerUpdateFrequency
+  ? parseInt(process.env.providerUpdateFrequency, 10)
+  : 60 * 60 * 1000
```

Comment:
Was this necessary in the end? I thought it was agreed to just make the request for the providers list on every request to SSR?

Reply:
Yes, we've decided to make the request for the providers list on every request to SSR. However, without this check, it was making more than one provider request per SSR. It was updating on the server and on the client, too.

Reply:
> it was making more than one provider request per SSR

Is this because multiple places request it? Is there somehow we could centralize it per request :thinking: Like adding it to the request context or something I guess?